### PR TITLE
Postmortem for 0.3.1

### LINKED ISSUE #73: Release 0.3.1
- [x] merge all outstanding PRs
  - [x] https://github.com/jupyrdf/ipyforcegraph/pull/72
- [x] ensure the versions have been bumped (check with `doit`)
- [x] ensure the CHANGELOG is up-to-date
  - [x] move the new release to the top of the stack
  - [x] missed things added to #63
- [x] validate on binder
  - [x] https://mybinder.org/v2/gh/jupyrdf/ipyforcegraph/26d054e?urlpath=lab%2Ftree%2Fexamples%2F_index.ipynb
- [x] validate on ReadTheDocs
  - [x] https://readthedocs.org/projects/ipyforcegraph/builds/19885363/
- [x] wait for a successful build of `main`
  - [x] https://github.com/jupyrdf/ipyforcegraph/actions/runs/4504879359/jobs/7929909571
- [x] download the `dist` archive and unpack somewhere (maybe a fresh `dist`)
- [x] create a new release through the GitHub UI
  - [x] paste in the relevant CHANGELOG entries
  - [x] upload the artifacts
- [x] actually upload to npm.com, pypi.org
  ```bash
  cd dist
  twine upload *.tar.gz *.whl
  npm login
  npm publish jupyrdf-jupyter-forcegraph-$VERSION.tgz
  npm logout
  ```
  - [x] https://pypi.org/project/ipyforcegraph/0.3.1/
  - [x] https://www.npmjs.com/package/@jupyrdf/jupyter-forcegraph/v/0.3.1 
- [x] postmortem
  - [x] handle `conda-forge` feedstock tasks
    - [x] https://github.com/conda-forge/ipyforcegraph-feedstock/pull/3
    - [x] https://github.com/conda-forge/ipyforcegraph-feedstock/runs/12236248402
  - [x] validate on binder via simplest-possible gists
    - [x] pip
      - [x] [Binder](https://mybinder.org/v2/git/https%3A%2F%2Fgist.github.com%2Fnrbgt%2F423ca19397505211424bc779dfb6a1d5/HEAD)
    - [x] conda 
      - [x] [Binder](https://mybinder.org/v2/git/https%3A%2F%2Fgist.github.com%2Fnrbgt%2F70d0e715220a3e566a669077efe3c32f/HEAD)
  - [x] bump to next development version
  - [x] bump the `CACHE_EPOCH`
  - [x] rebuild `yarn.lock`
  - [x] remove `.github/locks` and `doit lock`
  - [ ] update release procedures with lessons learned

### PR DESCRIPTION
**Pull request by nrbgt**
## Checklist

- [x] the PR target is correct
  - `main` for releases, docs-only fixes and post-release hot-fixes
  - `dev` for everything else
- [x] ran `doit lint` locally

## References

- fixes #73

## Code changes

- [x] update versions to `0.3.2`
  - [x] `CHANGELOG` (inc 0.3.1 for #62)
  - [x] `pyproject.toml`
  - [x] `package.json`
- [x] resolve `yarn.lock` (no change? :thinking: ) 
- [x] relock envs (no change, but not surprising)

## User-facing changes

- n/a

## Backwards-incompatible changes
- n/a

### CODE REVIEW

File: docs/environment.yml [hunk 1]
```diff
@@ -5,15 +5,19 @@ dependencies:
...
   - autodoc-traits
   - bqplot >=0.12.20
+  - comm <0.1.3
   - docutils >=0.19
   - doit-with-toml
   - flit >=3.7.1,<4
   - git
-  - ipywidgets >=8.0.4,<9
+  - ipykernel <6.22.0
+  - ipywidgets <8.0.5
```
Comment (sanbales) [inline docs/environment.yml:14]: are these two `ipywidgets` lines intentional?

File: .github/locks/osx-64_test_lab3.6_py3.11.conda.lock [hunk 2]
```diff
@@ -6,7 +6,7 @@
 #   - firefox =102
 #   - geckodriver
 #   - importnb
-#   - ipywidgets >=8.0.4,<9
+#   - ipywidgets >=8.0.0
```
Comment (sanbales) [inline .github/locks/osx-64_test_lab3.6_py3.11.conda.lock:9]: is this `<9` no longer valid?  I noticed it was still used the `pyproject.toml` file

**Review by sanbales — APPROVED** [review]
Tested the Behaviors nb in lite and everything seems copacetic, only question I had was if those `ipywidget` versions were intentional

File: docs/environment.yml (hunk 1 above)
Comment (nrbgt) [reply docs/environment.yml:14]: Yep, bottom pin is set in `_run.yml` and `lab3.5.yml` sets a top pin. not worried about `9` right now, let's deal with the breaking changes in `8.x +/ 1`

**Pull request merged**